c/tskit/tables.c

```diff
@@ -7818,13 +7838,19 @@ ancestor_mapper_run(ancestor_mapper_t *self)
                     goto out;
                 }
                 current_parent = parent;
+                if (node_time[current_parent] > self->oldest_node_time) {
+                    break;
+                }
+
                 start = j;
```

Comment:
I think we can make this a bit simpler - why do we have the last call to ancestor_mapper_process_parent_edges below? 

I think you should put the ``break`` after the ``start = j`` bit also, to keep the loop invariants correct.

Reply:
I guess it depends on what makes things simpler -- the latest push changes it so that process_parent_edges is only called inside the loop, but it means everything in the loop has to be put inside conditional statements.